golem_messages/message/tasks.py

```diff
@@ -234,31 +234,52 @@ def __init__(self, *args, **kwargs):
 
 @library.register(TASK_MSG_BASE + 1)
 class WantToComputeTask(ConcentEnabled, base.Message):
+    """
+    A computation Offer.
+
+    A Provider sends it directly to a Requestor as a response to the Requestor's
+    Demand (Task) in order to get work (SubTask ie. TaskToCompute) to do.
+
+    Attributes:
+        node_name:          Provider's node name
+        perf_index:         Provider's Performance index
+        max_resource_size:  Provider's Storage size available for computation
+        max_memory_size:    Provider's RAM
+        price:              Offered price in GNT "WEI" (10e-18)
+        num_subtasks:       How many subtasks Provider wants to work on
+                            (simultaneously); 1 by default
+        concent_enabled:    Provider's Concent status
+        extra_data:         additional required information about the Provider's
+                            environment. `golem-messages` should be
+                            intentionally agnostic with regards to the contents
+                            of this field.
+        provider_public_key:
+                            for signing and encryption
+        provider_ethereum_address:
+                            for transactions on ETH blockchain
+        task_header:        Demand; signed by a Requestor
+    """
     __slots__ = [
         'node_name',
         'perf_index',
         'max_resource_size',
         'max_memory_size',
-        'num_cores',
         'price',
-        'concent_enabled',  # Provider notifies requestor
-                            # about his concent status
-
-        'extra_data',       # used to specify additional required
-                            # information about the provider's environment.
-                            # `golem-messages` should be intentionally agnostic
-                            # with regards to the contents of this field.
-
-        'provider_public_key',  # key used for msg signing and encryption
-        'provider_ethereum_public_key',  # used for transactions on blockchain
-        'task_header',  # TaskHeader, should be signed by requestor
+        'num_subtasks',
+        'concent_enabled',
+        'extra_data',
+        # TODO: `provider` prefix is redundant; all above fields refers Provider
```

Comment:
"explicit is better than implicit"

```suggestion
```

Reply:
agreed, then lets consequently name all others with prefix `provider_`
this is even more valid in case of TTC, where some fields are from requestor and some from provider afaik